internal/consensus/reactor.go

```diff
@@ -557,6 +560,14 @@ func (conR *Reactor) updateRoundStateRoutine() {
 	}
 }
 
+func (conR *Reactor) updateRoundStateNoCsLock() {
+	rs := conR.conS.getRoundState()
+	conR.rsMtx.Lock()
+	conR.rs = rs
+	conR.initialHeight = conR.conS.state.InitialHeight
```

Comment:
```suggestion
	conR.conS.mtx.RLock()
	conR.initialHeight = conR.conS.state.InitialHeight
	conR.conS.mtx.RULock()
```

how come we don't need a mutex to access state anymore?

Reply:
This is in the `noCsLock` function!